### PR TITLE
Add FlutterView and ViewManager and hook them into the ResidentRunner.

### PR DESCRIPTION
**Pull request by johnmccutchan**
- [x] Refactor view support into separate classes.
- [x] Make the hot runner grab the main view and call runFromSource on it.
- [x] Remove Device.needsDevFS (because it is always true).

@devoncarew @turnidge

### CODE REVIEW

File: packages/flutter_tools/lib/src/view.dart [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import 'observatory.dart';
+
+/// Peered to a Android/iOS FlutterView widget on a device.
+class FlutterView {
+  FlutterView(this.viewId, this.viewManager);
+
+  final String viewId;
+  String _uiIsolateId;
+  String get uiIsolateId => _uiIsolateId;
+  final ViewManager viewManager;
```
Comment (turnidge) [inline packages/flutter_tools/lib/src/view.dart]: Declare this right after viewId?

File: packages/flutter_tools/lib/src/view.dart [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+       serviceProtocol.onIsolateEvent.listen((Event event) {
+     // TODO(johnmccutchan): Listen to the debug stream and catch initial
+     // launch errors.
+     if (event.kind == 'IsolateRunnable') {
+       printTrace('Isolate is runnable.');
+       completer.complete(null);
+     }
+    });
+    await serviceProtocol.runInView(viewId,
+                                   entryPath,
```
Comment (turnidge) [inline packages/flutter_tools/lib/src/view.dart]: Check indentation.

**Comment by turnidge** [comment]
lgtm w/ 2 comments.

File: packages/flutter_tools/lib/src/view.dart (hunk 1 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/view.dart]: Done.

File: packages/flutter_tools/lib/src/view.dart (hunk 2 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/view.dart]: Fixed.

**Pull request merged**